File: bridges/voiceover/Sources/VoiceOverBridgeAdapters/Ports/Transport.swift
// ROLE: adapter seam for a raw byte pipe, and the signal that its poll window elapsed.
// USED BY: JsonLinesChannel.
// IMPLEMENTED BY: SocketTransport and FakeTransport.

import Foundation

/// Not a failure: it hands control back so the caller can check deadlines or notice it was asked to stop.
public struct PollTimeout: Error {
	public init() {}
}

public protocol Transport: AnyObject {
	/// The next chunk; empty at end of stream, and `PollTimeout` when idle.
	func receive() throws -> Data

	func sendAll(_ data: Data) throws

	func close()
}

File: bridges/nvda/addon/globalPlugins/nvdaMcpBridge/adapters/ports/transport.py
# nvdaMcpBridge adapters -- the Transport seam.
# Copyright (C) 2026 Marlon Brandao de Sousa. GPL-2. See COPYING.txt.
# ROLE: adapter seam for a raw byte pipe.
# USED BY: adapters/json_lines_channel.py.
# IMPLEMENTED BY: adapters/socket_transport.py, adapters/named_pipe_transport.py and tests/fakes/transport.py.

from __future__ import annotations

from abc import ABC, abstractmethod


class Transport(ABC):
	@abstractmethod
	def recv(self) -> bytes:
		"""Next chunk; ``b""`` at EOF; raises TimeoutError when idle."""

	@abstractmethod
	def sendall(self, data: bytes) -> None: ...

	@abstractmethod
	def close(self) -> None: ...
